File: ios/GuideStreamTV/Views/BrowseSortSheet.swift
//
//  BrowseSortSheet.swift
//  GuideStreamTV
//
//  Sort is a separate, compact control from filtering: one choice, applied on
//  tap, no confirm step. Four options do not deserve a full-screen sheet.
//

import SwiftUI

struct BrowseSortSheet: View {
    let sort: BrowseSort
    var onSelect: (BrowseSort) -> Void

    @Environment(\.dismiss) private var dismiss

    var body: some View {
        VStack(spacing: 0) {
            GsSheetHeader(title: "Sort by")
                .padding(.horizontal, 8)

            VStack(spacing: 0) {
                ForEach(BrowseSort.allCases, id: \.rawValue) { option in
                    Button {
                        onSelect(option)
                        dismiss()
                    } label: {
                        HStack(spacing: 12) {
                            ZStack {
                                Circle()
                                    .stroke(
                                        option == sort ? Color.orange : Color.white.opacity(0.35),
                                        lineWidth: 1.5
                                    )
                                    .frame(width: 20, height: 20)
                                if option == sort {
                                    Circle().fill(Color.orange).frame(width: 11, height: 11)
                                }
                            }

                            Text(option.label)
                                .scaledFont(size: 15)
                                .foregroundStyle(.white)

                            Spacer()
                        }
                        .padding(.vertical, 13)
                        .contentShape(Rectangle())
                    }
                    .buttonStyle(.plain)

                    if option != BrowseSort.allCases.last {
                        Rectangle()
                            .fill(Color.white.opacity(0.07))
                            .frame(height: 1)
                    }
                }
            }
            .padding(.horizontal, 18)

            Spacer(minLength: 0)
        }
        .sheetSurface(.base)
        .presentationDetents([.height(320)])
        .preferredColorScheme(.dark)
    }
}

File: android/app/src/main/java/com/rork/guidestreamtvandroid/ui/browse/BrowseSortSheet.kt
package com.rork.guidestreamtvandroid.ui.browse

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.rork.guidestreamtvandroid.data.models.BrowseSort
import com.rork.guidestreamtvandroid.ui.components.GsSheetDragHandle
import com.rork.guidestreamtvandroid.ui.theme.BrandOrange
import com.rork.guidestreamtvandroid.ui.theme.SheetLevel
import com.rork.guidestreamtvandroid.ui.theme.SheetSurfaceBase
import com.rork.guidestreamtvandroid.ui.theme.sheetTopInset

/**
 * Sort is a separate, compact control from filtering: one choice, applied on
 * tap, no confirm step. Four options do not deserve a full-height sheet.
 * Mirrors iOS `BrowseSortSheet`.
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun BrowseSortSheet(
    sort: BrowseSort,
    onSelect: (BrowseSort) -> Unit,
    onDismiss: () -> Unit,
) {
    val sheetState = rememberModalBottomSheetState()

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        containerColor = SheetSurfaceBase,
        scrimColor = Color.Black.copy(alpha = 0.60f),
        tonalElevation = 0.dp,
        dragHandle = { GsSheetDragHandle(level = SheetLevel.Base) },
        contentWindowInsets = { sheetTopInset() },
    ) {
        Column(modifier = Modifier.fillMaxWidth().padding(horizontal = 18.dp)) {
            Text(
                text = "Sort by",
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White,
                modifier = Modifier.padding(top = 8.dp, bottom = 10.dp),
            )

            BrowseSort.entries.forEach { option ->
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable(
                            interactionSource = remember { MutableInteractionSource() },
                            indication = null,
                        ) {
                            onSelect(option)
                            onDismiss()
                        }
                        .padding(vertical = 13.dp),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(12.dp),
                ) {
                    Box(
                        modifier = Modifier
                            .size(20.dp)
                            .clip(CircleShape)
                            .border(
                                width = 1.5.dp,
                                color = if (option == sort) BrandOrange else Color.White.copy(alpha = 0.35f),
                                shape = CircleShape,
                            ),
                        contentAlignment = Alignment.Center,
                    ) {
                        if (option == sort) {
                            Box(
                                modifier = Modifier
                                    .size(11.dp)
                                    .clip(CircleShape)
                                    .background(BrandOrange)
                            )
                        }
                    }
                    Text(option.label, fontSize = 15.sp, color = Color.White)
                }

                if (option != BrowseSort.entries.last()) {
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(1.dp)
                            .background(Color.White.copy(alpha = 0.07f))
                    )
                }
            }

            Spacer(modifier = Modifier.height(30.dp))
        }
    }
}
